manifests/init.pp

```diff
@@ -1,151 +1,59 @@
-  } else {
-    $use_fqdn_real = $use_fqdn
-  }
-
-  # validate type and convert string to boolean if necessary
-  if is_string($purge_hosts) {
-    $purge_hosts_enabled = str2bool($purge_hosts)
-  } else {
-    $purge_hosts_enabled = $purge_hosts
-  }
-
-  if $ipv4_localhost_enabled == true {
-    $localhost_ensure     = 'present'
-    $localhost_ip         = '127.0.0.1'
-    $my_localhost_aliases = $localhost_aliases
-  } else {
-    $localhost_ensure     = 'absent'
-    $localhost_ip         = '127.0.0.1'
-    $my_localhost_aliases = undef
-  }
-
-  if $ipv6_localhost_enabled == true {
-    $localhost6_ensure     = 'present'
-    $localhost6_ip         = '::1'
-    $my_localhost6_aliases = $localhost6_aliases
-  } else {
-    $localhost6_ensure     = 'absent'
-    $localhost6_ip         = '::1'
-    $my_localhost6_aliases = undef
-  }
-
-  if !is_string($my_localhost_aliases) and !is_array($my_localhost_aliases) {
-    fail('hosts::localhost_aliases must be a string or an array.')
-  }
-
-  if !is_string($my_localhost6_aliases) and !is_array($my_localhost6_aliases) {
-    fail('hosts::localhost6_aliases must be a string or an array.')
-  }
-
-  if $fqdn_entry_enabled == true {
+  if $fqdn_entry == true {
     $fqdn_ensure          = 'present'
-    $my_fqdn_host_aliases = $fqdn_host_aliases
-    $fqdn_ip              = $::ipaddress
+    $_fqdn_host_aliases = $fqdn_host_aliases
   } else {
     $fqdn_ensure          = 'absent'
-    $my_fqdn_host_aliases = []
-    $fqdn_ip              = $::ipaddress
+    $_fqdn_host_aliases = undef
   }
 
-  Host {
-    target => $target,
+  $host_defaults = {
+    'target' => $target,
   }
 
-  host { 'localhost':
-    ensure => 'absent',
+  host { $::fqdn:
+    ensure       => $fqdn_ensure,
+    host_aliases => $_fqdn_host_aliases,
+    ip           => $fqdn_ip,
+    *            => $host_defaults,
   }
 
-  host { 'localhost.localdomain':
-    ensure       => $localhost_ensure,
-    host_aliases => $my_localhost_aliases,
-    ip           => $localhost_ip,
-  }
-
-  host { 'localhost6.localdomain6':
-    ensure       => $localhost6_ensure,
-    host_aliases => $my_localhost6_aliases,
-    ip           => $localhost6_ip,
+  resources { 'host':
+    purge => $purge_hosts,
   }
 
-  if $use_fqdn_real == true {
-    @@host { $::fqdn:
-      ensure       => $fqdn_ensure,
-      host_aliases => $my_fqdn_host_aliases,
-      ip           => $fqdn_ip,
+  if $host_entries != undef {
+    if $target == undef {
+      $host_entries_merged = $host_entries
+    } else {
+      $host_entries_merged = $host_defaults + $host_entries
     }
+    $_host_entries = delete($host_entries_merged, $::fqdn)
 
-    case $collect_all_real {
-      # collect all the exported Host resources
-      true:  {
-        Host <<| |>>
-      }
-      # only collect the exported entry above
-      default: {
-        Host <<| title == $::fqdn |>>
+    $_host_entries.each |$host, $params| {
+      host { $host:
+        * => $params,
       }
```

Comment:
IF you are going to use $host_defaults instead of Host, you need to merge $params and $host_defaults. Alternatively, go back to the "Host" setup.

Reply:
Newest commit has it merged. Good eye that I missed that :)